enclave/seal.c

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openenclave/enclave.h>
+#include <openenclave/corelibc/stdlib.h>
+#include <openenclave/internal/crypto/gcm.h>
+#include <openenclave/internal/raise.h>
+#include <openenclave/internal/safecrt.h>
+
+static const uint8_t _iv[12] = { 0 };
+
+oe_result_t oe_seal(
+    const uint8_t* key_info,
+    size_t key_info_size,
```

Comment:
How would crypto-agility work?  For example, if we want to allow using say an AES-CCM Elliptic Curve Cryptography (ECC) Cipher Suite, or whatever else, instead of GCM to do AEAD, would you simply pass in different key_info?   Or do we need an additional argument to say to use GCM here?